docs/spec/auth/vesting.md

```diff
@@ -1,159 +1,138 @@
+during genesis with a starting balance `X` coins and a vesting end time `T`.
 
-### Implementation
+The owner of this account should be able to delegate to validators
+and vote with locked coins, however they cannot send locked coins to other
+accounts until those coins have been unlocked.
 
-##### Vesting Account implementation
+In addition, the vesting account should also be able to spend any coins it
+receives from other users. Thus, the bank module's `MsgSend` handler should
+error if a vesting account is trying to send an amount that exceeds their
+unlocked coin amount.
 
-NOTE:  `Now = ctx.BlockHeader().Time`
+## Vesting Account Definition
 
 ```go
 type VestingAccount interface {
     Account
-    AssertIsVestingAccount() // existence implies that account is vesting.
+    AssertIsVestingAccount() // existence implies that account is vesting
 
-    // Calculates amount of coins that can be sent to other accounts given the current time
+    // Calculates the amount of coins that can be sent to other accounts given
+    // the current time.
     SendableCoins(sdk.Context) sdk.Coins
 }
 
-// Implements Vesting Account
-// Continuously vests by unlocking coins linearly with respect to time
+// ContinuousVestingAccount implements the Vesting Account interface. It
+// continuously vests by unlocking coins linearly with respect to time.
 type ContinuousVestingAccount struct {
     BaseAccount
-    OriginalVestingCoins sdk.Coins // Coins in account on Initialization
-    ReceivedCoins        sdk.Coins // Coins received from other accounts
-    SentCoins            sdk.Coins // Coins sent to other accounts
 
-    // StartTime and EndTime used to calculate how much of OriginalCoins is unlocked at any given point
+    OriginalVesting   sdk.Coins // coins in account upon initialization
+    DelegatedVesting  sdk.Coins // coins that vesting and delegated
+    DelegatedFree     sdk.Coins // coins that are vested and delegated
+
+    // StartTime and EndTime are used to calculate how much of OriginalVesting
+    // is unlocked at any given point.
     StartTime time.Time
     EndTime   time.Time
 }
+```
 
-// Uses time in context to calculate total unlocked coins
-SendableCoins(vacc ContinuousVestingAccount, ctx sdk.Context) sdk.Coins:
-    
-    // Coins unlocked by vesting schedule
-    unlockedCoins := ReceivedCoins - SentCoins + OriginalVestingCoins * (Now - StartTime) / (EndTime - StartTime)
-
-    // Must still check for currentCoins constraint since some unlocked coins may have been delegated.
-    currentCoins := vacc.BaseAccount.GetCoins()
+## Vesting Account Implementation
 
-    // min will return sdk.Coins with each denom having the minimum amount from unlockedCoins and currentCoins
-    return min(unlockedCoins, currentCoins)
+Given a vesting account, we may further define the following:
 
+```
+OV = OriginalVesting (constant)
+V = Percentage of OV that is still vesting (derived by OV and the start/end times)
```

Comment:
This isn't consistent, sometimes `V` seems to be a percentage and sometimes an amount of coins (the amount of coins yet to vest). We should choose one or the other.

Reply:
You're 100% correct. I think it should be # of coins, not %. I'll update.